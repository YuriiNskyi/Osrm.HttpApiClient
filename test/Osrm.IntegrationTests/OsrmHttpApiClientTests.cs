using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Osrm.Examples;
using Osrm.HttpApiClient;

namespace Osrm.IntegrationTests
{
    [UseApprovalSubdirectory("Approvals")]
    [UseReporter(typeof(VisualStudioReporter))]
    public class OsrmHttpApiClientTests
    {
        private HttpClient _httpClient;
        private OsrmHttpApiClient _osrmClient;

        [SetUp]
        public void Setup()
        {
            _httpClient = new HttpClient
            {
                BaseAddress = new Uri("http://router.project-osrm.org"),
            };

            _osrmClient = new OsrmHttpApiClient(_httpClient);
        }

        [TearDown]
        public void Teardown()
        {
            _osrmClient = null;
            _httpClient.Dispose();
        }

        [Test]
        public async Task Full_GeoJson_route_query()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Full_GeoJson_route_query);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Full_Polyline_route_query()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Full_Polyline_route_query);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Query_on_Berlin_with_three_coordinates()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Query_on_Berlin_with_three_coordinates);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Query_on_Berlin_excluding_the_usage_of_motorways()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Query_on_Berlin_excluding_the_usage_of_motorways);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Query_on_Berlin_using_polyline()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Query_on_Berlin_using_polyline);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Querying_nearest_three_snapped_locations()
        {
            var response = await _osrmClient.GetNearestAsync(ProjectOsrmOrgDocs.Querying_nearest_three_snapped_locations);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_matrix()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_matrix);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_1x3_duration_matrix()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_1x3_duration_matrix);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task An_asymmetric_3x2_duration_matrix_with_from_the_polyline_encoded_locations()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.An_asymmetric_3x2_duration_matrix_with_from_the_polyline_encoded_locations);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_duration_matrix()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_duration_matrix);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_duration_matrix_for_CH()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_duration_matrix_for_CH);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_duration_matrix_and_a_3x3_distance_matrix_for_CH()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_duration_matrix_and_a_3x3_distance_matrix_for_CH);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Round_trip_in_Berlin_with_three_stops()
        {
            var response = await _osrmClient.GetTripAsync(ProjectOsrmOrgDocs.Round_trip_in_Berlin_with_three_stops);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Round_trip_in_Berlin_with_four_stops_starting_at_the_first_stop_ending_at_the_last()
        {
            var response = await _osrmClient.GetTripAsync(ProjectOsrmOrgDocs.Round_trip_in_Berlin_with_four_stops_starting_at_the_first_stop_ending_at_the_last);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }
    }
}