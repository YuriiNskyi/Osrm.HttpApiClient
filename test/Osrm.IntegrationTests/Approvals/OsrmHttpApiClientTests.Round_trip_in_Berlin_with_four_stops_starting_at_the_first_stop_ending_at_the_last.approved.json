{
    "waypoints": [
        {
            "trips_index": 0,
            "waypoint_index": 0,
            "nodes": [],
            "name": "Friedrichstra\u00DFe",
            "location": [
                13.388798,
                52.517033
            ],
            "distance": 4.231666,
            "hint": "qqDXgYYvApMYAAAAJgAAAAAAAAAAAAAASjFaQU1xpUEAAAAAAAAAAAwAAAATAAAAAAAAAAAAAADXFgEA_kvMAKlYIQM8TMwArVghAwAA7wp2gVsJ"
        },
        {
            "trips_index": 0,
            "waypoint_index": 1,
            "nodes": [],
            "name": "Torstra\u00DFe",
            "location": [
                13.397631,
                52.529432
            ],
            "distance": 2.789393,
            "hint": "IkJKgCy_9JIGAAAACgAAAAAAAAB2AAAAW7-PQOKcyEAAAAAApq6DQgYAAAAKAAAAAAAAAHYAAADXFgEAf27MABiJIQOCbswA_4ghAwAAXwV2gVsJ"
        },
        {
            "trips_index": 0,
            "waypoint_index": 2,
            "nodes": [],
            "name": "Platz der Vereinten Nationen",
            "location": [
                13.428554,
                52.523239
            ],
            "distance": 2.226595,
            "hint": "JUJKgP___38fAAAAUQAAACYAAAAeAAAAsowKQkpQX0Lx6yZCvsQGQh8AAABRAAAAJgAAAB4AAADXFgEASufMAOdwIQNL58wA03AhAwMAvxB2gVsJ"
        },
        {
            "trips_index": 0,
            "waypoint_index": 3,
            "nodes": [],
            "name": "",
            "location": [
                13.418364,
                52.523292
            ],
            "distance": 15.539673,
            "hint": "P770kv___3_XAAAADQEAADAAAAAPAAAA3dqzQos3skHbKaNB3lbDQNcAAAANAQAAMAAAAA8AAADXFgEAfL_MABxxIQM7wMwAz3AhAwQAHwJ2gVsJ"
        }
    ],
    "trips": [
        {
            "distance": 10082.1,
            "duration": 1235,
            "geometry": {
                "coordinates": [
                    [
                        13.388798,
                        52.517033
                    ],
                    [
                        13.388625,
                        52.518883
                    ],
                    [
                        13.387885,
                        52.522526
                    ],
                    [
                        13.387215,
                        52.527164
                    ],
                    [
                        13.393668,
                        52.528492
                    ],
                    [
                        13.397631,
                        52.52943
                    ],
                    [
                        13.403888,
                        52.529682
                    ],
                    [
                        13.411389,
                        52.528324
                    ],
                    [
                        13.41432,
                        52.527508
                    ],
                    [
                        13.415657,
                        52.52677
                    ],
                    [
                        13.417166,
                        52.528458
                    ],
                    [
                        13.421348,
                        52.52873
                    ],
                    [
                        13.424085,
                        52.528084
                    ],
                    [
                        13.424993,
                        52.52807
                    ],
                    [
                        13.425184,
                        52.527885
                    ],
                    [
                        13.427263,
                        52.527042
                    ],
                    [
                        13.43036,
                        52.525063
                    ],
                    [
                        13.430413,
                        52.52479
                    ],
                    [
                        13.429678,
                        52.52327
                    ],
                    [
                        13.428554,
                        52.52324
                    ],
                    [
                        13.426375,
                        52.5232
                    ],
                    [
                        13.425475,
                        52.52334
                    ],
                    [
                        13.429315,
                        52.525738
                    ],
                    [
                        13.430227,
                        52.525196
                    ],
                    [
                        13.430413,
                        52.52479
                    ],
                    [
                        13.429705,
                        52.52343
                    ],
                    [
                        13.429447,
                        52.521217
                    ],
                    [
                        13.428426,
                        52.519127
                    ],
                    [
                        13.427889,
                        52.51891
                    ],
                    [
                        13.427126,
                        52.51897
                    ],
                    [
                        13.425964,
                        52.51927
                    ],
                    [
                        13.416715,
                        52.52206
                    ],
                    [
                        13.417399,
                        52.52268
                    ],
                    [
                        13.417606,
                        52.522594
                    ],
                    [
                        13.418364,
                        52.523293
                    ],
                    [
                        13.418594,
                        52.523506
                    ],
                    [
                        13.418392,
                        52.523586
                    ],
                    [
                        13.418667,
                        52.523888
                    ],
                    [
                        13.420239,
                        52.52522
                    ],
                    [
                        13.415682,
                        52.527176
                    ],
                    [
                        13.411596,
                        52.523785
                    ],
                    [
                        13.409986,
                        52.52275
                    ],
                    [
                        13.40482,
                        52.520157
                    ],
                    [
                        13.403043,
                        52.519108
                    ],
                    [
                        13.401966,
                        52.518673
                    ],
                    [
                        13.399268,
                        52.517746
                    ],
                    [
                        13.388779,
                        52.517155
                    ],
                    [
                        13.388798,
                        52.517033
                    ]
                ]
            },
            "weight": 1237.4,
            "weight_name": "routability",
            "legs": [
                {
                    "distance": 1884.8,
                    "duration": 251.5,
                    "weight": 252.7,
                    "summary": "",
                    "steps": [],
                    "annotations": null
                },
                {
                    "distance": 2839.1,
                    "duration": 363.5,
                    "weight": 363.5,
                    "summary": "",
                    "steps": [],
                    "annotations": null
                },
                {
                    "distance": 2442,
                    "duration": 280.2,
                    "weight": 280.2,
                    "summary": "",
                    "steps": [],
                    "annotations": null
                },
                {
                    "distance": 2916.2,
                    "duration": 339.8,
                    "weight": 341,
                    "summary": "",
                    "steps": [],
                    "annotations": null
                }
            ]
        }
    ],
    "code": "Ok",
    "message": null
}